ViewController.swift

```diff
@@ -7,13 +7,100 @@
+        picker.allowsEditing = true
+        picker.delegate = self
+        present(picker, animated: true)
+    }
+    
+    func imagePickerController(_ picker: UIImagePickerController, didFinishPickingMediaWithInfo info: [UIImagePickerController.InfoKey : Any]) {
+        guard let image = info[.editedImage] as? UIImage else { return }
+        let imageName = UUID().uuidString
+        let imagepath = getDocumentsDirectory().appendingPathComponent(imageName)
+        
+        if let jpegData = image.jpegData(compressionQuality: 0.8) {
+            try? jpegData.write(to: imagepath)
+        }
+        let person = Person(name: "Unknown", image: imageName)
+        people.append(person)
+        collectionView.reloadData()
+        dismiss(animated: true)
+    }
+    
+    private func getDocumentsDirectory() -> URL {
+        let paths = FileManager.default.urls(for: .documentDirectory, in: .userDomainMask)
+        return paths[0]
+    }
+    
+    override func collectionView(_ collectionView: UICollectionView, didSelectItemAt indexPath: IndexPath) {
+        let person = people[indexPath.item]
+        
+        let alertController = UIAlertController(title: nil, message: "Please Select an Option", preferredStyle: .actionSheet)
+        
+        alertController.addAction(UIAlertAction(title: "Rename", style: .default, handler: { (_) in
+            
+            let renameAlertController = UIAlertController(title: "Rename", message: "What's the new name?", preferredStyle: .alert)
+            renameAlertController.addTextField()
+            
+            renameAlertController.addAction(UIAlertAction(title: "Rename", style: .default) { [weak self, weak renameAlertController] _ in
+                guard let self = self else { return }
+                guard let newName = renameAlertController?.textFields?[0].text else { return }
+                person.name = newName
+                
+                self.collectionView.reloadData()
+            })
+            
+            renameAlertController.addAction(UIAlertAction(title: "Cancel", style: .cancel, handler: { [weak renameAlertController] _ in
+                renameAlertController?.dismiss(animated: true)
+            }))
+            
+            self.present(renameAlertController, animated: true)
+        }))
+        
+        alertController.addAction(UIAlertAction(title: "Delete", style: .destructive, handler: { [weak self, weak alertController] _ in
+            guard let self = self else { return }
+            self.people.remove(at: indexPath.item)
+            alertController?.dismiss(animated: true)
+            self.collectionView.reloadData()
+        }))
+        
+        alertController.addAction(UIAlertAction(title: "Cancel", style: .default, handler: { [weak alertController] _ in
+            alertController?.dismiss(animated: true)
+        }))
+        
+        
+        self.present(alertController, animated: true, completion: {
```

Comment:
If you take a look at the present method signature you'll see that completion has default value.
```
func present(_ viewControllerToPresent: UIViewController, animated flag: Bool,  completion: (() -> Void)? = nil)
```

In cases like this, you can omit the value, if you're ok with the default one.

So, here, this can be simplified to 
```suggestion
        self.present(alertController, animated: true)
```

Reference: https://developer.apple.com/documentation/uikit/uiviewcontroller/1621380-present

Reply:
Fixed!